### ISSUE TITLE
Improve dbm file version control

### ISSUE THREAD

**Issue by rkhaotix** [pgmodeler/pgmodeler#1637]
Hi!

I'm attaching to this old issue with a question. @rkhaotix did you get a chance to plan the new feature for reverse engineering?

I recently started using pgmodeler with my team. What saw as advantage is to keep the DB model as close to code as possible, so I proposed saving the model in git repository. At first it looks ok, because if you introduce new feature, you can update the model (keeping documentation up to date effortlessly!). Then using dockerized version of postgres and migrations implemented in project and pgmodeler's diff tool you can generate next migration and save it in project. Neat.

Looks great, but it works only if only one person is working on a task requiring change in the model. Otherwise (and that's normal case) the model gets conflicts and only the first person gets to merge their changes... 

Even if the model was not stored in git, but we'd keep it on a shared drive - the problem of working in parallel on 2 different features is still there (just the merge conflict is gone). There can be only one version of the model at a time because otherwise there is no simple way of merging work.

Does anyone have idea how to use pgmodeler to work on separate features and then merge the models?

_Originally posted by @mkocztorz in https://github.com/pgmodeler/pgmodeler/issues/1393#issuecomment-1013048184_

**Comment by mkocztorz** [comment]
Hi @rkhaotix! Any news about this? It's really a show stopper in the long run.

**Comment by rkhaotix** [comment]
Unfortunately, no.

But I will try to work on this during pgModeler 1.0 development.

**Comment by rkhaotix** [comment]
@mkocztorz Hi! Long time since we don't talk about this issue (and #1585 too).

Well, I guess I came up with a possible solution to this problem. I was thinking of creating a new way to write pgModeler models to disk in order to make them more SCM-friendly. 

The idea is to write individual XML files (see below) for each object so during a merge/diff in git you can track the changes to be performed more efficiently without corrupting the entire database model as happens to a single dbm file.

Anyway, the new arrangement can be constructed like the following (with some variations but the idea is the same):

```
mydbmodel    --> Instead of a single .dbm file the model would be stored in a directory.
|
+--- index.xml --> This is the main file and the entry point of the loading process. 
     |           It'll store the correct order in which pgModeler should load the other 
     |           files to recreate the objects.
     |
     +--- public_table1.xml
     |
     +--- public_table1_trigger1.xml
     |
     +--- schema_a_table2.xml
     |
     +--- schema_a_table2_index1.xml
     |
     +--- relationship_table1_table2.xml
``` 

I didn't test it yet, but I think that arranging the files in the above schema can help you to work with an SCM in parallel with other people on your team. I see some downsides with this approach depending on the size of your database model, you can have hundreds or even thousands of files within a directory.

I'm planning to add this feature to pgModeler 1.1.0 since I'm currently working on 1.0.0-beta and focusing on bug fixes until the release of the stable version.

Anyway, I want to hear you. Feel free to give suggestions about this. ;)

**Comment by dekiesel** [comment]
Just to be sure since this is marked as completed: This feature isn't available yet but will be in 1.1.0? :) Thanks!

**Comment by rkhaotix** [comment]
@dekiesel This feature is already available in form of a plug-in which is distributed with the paid version of the tool.

https://pgmodeler.io/support/docs/822-split-database-model-format?v=1.1.0
